### PR TITLE
Implementation of ternary search

### PR DESCRIPTION
**Pull request by singloon**
Implementation of ternary search via TDD #116. Please can you review @fengyuanyang

### CODE REVIEW

**Review by fengyuanyang — COMMENTED** [review]
hi , can you have a look on it

File: src/main/java/com/ordestiny/tdd/algorithms/TernarySearch.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                high = mid1 - 1;
+            } else {
+                low = mid1 + 1;
+                high = mid2 - 1;
+            }
+        }
+
+        if (!isSorted(array)) {
+            throw new IllegalArgumentException();
+        }
```
[the remark this answers is not in the document]
Comment (singloon) [reply src/main/java/com/ordestiny/tdd/algorithms/TernarySearch.java]: I've moved the sorted check to the start

[the remark this answers is not in the document]
Comment (fengyuanyang) [reply src/main/java/com/ordestiny/tdd/algorithms/TernarySearch.java]: Thanks 👍 
LGTM now

**Review by fengyuanyang — APPROVED** [review]
thanks for this commit and update on .gitignore

**Pull request merged**